kitsune/wiki/views.py

```diff
@@ -127,6 +130,28 @@ def document(request, document_slug, template=None, document=None):
 
     any_localizable_revision = doc.revisions.filter(is_approved=True,
                                                     is_ready_for_localization=True).exists()
+    # Find and show the defined fallback locale rather than the English version of the document
+    # The fallback locale is defined based on the ACCEPT_LANGUAGE header,
+    # site-wide locale mapping and custom fallback locale
+    # The custom fallback locale is defined in the FALLBACK_LOCALES array in
+    # kitsune/wiki/config.py. See bug 800880 for more details
+    if fallback_reason == 'no_translation':
+        fallback_locale = get_fallback_locale(doc, request)
+
+        # If a fallback locale is defined, show the document in that locale.
+        if fallback_locale is not None:
+            # Get the fallback Locale and show doc in the locale
+            translation = doc.translated_to(fallback_locale)
+            doc = Document.objects.get(locale=fallback_locale,
+                                       slug=translation.slug)
```

Comment:
This has the same problem you're other PR did. You're fetching the document from the DB twice in a row.